x-pack/plugins/apm/server/routes/services/route.ts

```diff
@@ -274,7 +275,12 @@ const serviceMetadataDetailsRoute = createApmServerRoute({
     import('./get_service_metadata_details').ServiceMetadataDetails
   > => {
     const setup = await setupRequest(resources);
-    const { params, context, plugins } = resources;
+    const { plugins, context } = resources;
+    const infraMetricsClient = await createInfraMetricsClient({
+      infraPlugin: plugins.infra,
+      context,
+    });
```

Comment:
What about just passing `resources` as arg:

```suggestion
    const infraMetricsClient = await createInfraMetricsClient(resources);
```

Reply:
I leave like this, `infraPlugin` is not part of the resources, so I will need to extracted somewhere else